src/Misc/layoutroot/config.sh

```diff
@@ -20,30 +20,30 @@ then
 
     ldd ./bin/libcoreclr.so | grep 'not found'
     if [ $? -eq 0 ]; then
-        echo "Dependencies is missing for Dotnet Core 2.1"
-        echo "Execute ./bin/installdependencies.sh to install any missing Dotnet Core 2.1 dependencies."
+        echo "Dependencies is missing for .NET Core 3.1"
+        echo "Execute ./bin/installdependencies.sh to install any missing dependencies."
         exit 1
     fi
 
     ldd ./bin/System.Security.Cryptography.Native.OpenSsl.so | grep 'not found'
     if [ $? -eq 0 ]; then
-        echo "Dependencies is missing for Dotnet Core 2.1"
-        echo "Execute ./bin/installdependencies.sh to install any missing Dotnet Core 2.1 dependencies."
+        echo "Dependencies missing for .NET Core 3.1"
+        echo "Execute ./bin/installdependencies.sh to install any missing dependencies."
         exit 1
     fi
 
     ldd ./bin/System.IO.Compression.Native.so | grep 'not found'
     if [ $? -eq 0 ]; then
-        echo "Dependencies is missing for Dotnet Core 2.1"
-        echo "Execute ./bin/installdependencies.sh to install any missing Dotnet Core 2.1 dependencies."
+        echo "Dependencies missing for .NET Core 3.1"
+        echo "Execute ./bin/installdependencies.sh to install any missing dependencies."
         exit 1
     fi
 
     ldd ./bin/System.Net.Http.Native.so | grep 'not found'
     if [ $? -eq 0 ]; then
-        echo "Dependencies is missing for Dotnet Core 2.1"
-        echo "Execute ./bin/installdependencies.sh to install any missing Dotnet Core 2.1 dependencies."
-        exit 1
+        echo "Warning: on some platforms, libcurl3 is required."
```

Comment:
Can we detect platform here and still `exit 1` on those?

Reply:
We could. It would mirror the logic in all the `Program.cs`s (look for "linux" and either "arm" or "arm64") but in shell script instead of C#.

After this change ships, we're going to remove the dependency on libcurl3 altogether (well, possibly with a backcompat knob). Tommy wanted to do that as a separate change to reduce risk. Given that the whole block of code will go away when that happens, I'm not sure if it's worth reimplementing the logic here.

Reply:
Makes sense to me